golang/arguments.go

```diff
@@ -0,0 +1,103 @@
+package tchannel
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+)
+
+// An Input is able to read an argument from a call body
```

Comment:
Ah good call. And yea doesn't have to be now, just something we should strive for in the long term.